tools/msg/nes/msgdisNES.py

```diff
@@ -0,0 +1,300 @@
+            0xE3: ("CMD_SPIDER_HOUSE_MASK_CODE_3", self.decode_cmd_no_arg),
+            0xE4: ("CMD_SPIDER_HOUSE_MASK_CODE_4", self.decode_cmd_no_arg),
+            0xE5: ("CMD_SPIDER_HOUSE_MASK_CODE_5", self.decode_cmd_no_arg),
+            0xE6: ("CMD_SPIDER_HOUSE_MASK_CODE_6", self.decode_cmd_no_arg),
+            0xE7: ("CMD_HOURS_UNTIL_MOON_CRASH", self.decode_cmd_no_arg),
+            0xE8: ("CMD_TIME_UNTIL_NEW_DAY", self.decode_cmd_no_arg),
+            0xF0: ("CMD_HS_POINTS_BANK_RUPEES", self.decode_cmd_no_arg),
+            0xF1: ("CMD_HS_POINTS_UNK_1", self.decode_cmd_no_arg),
+            0xF2: ("CMD_HS_POINTS_FISHING", self.decode_cmd_no_arg),
+            0xF3: ("CMD_HS_TIME_BOAT_ARCHERY", self.decode_cmd_no_arg),
+            0xF4: ("CMD_HS_TIME_HORSE_BACK_BALLOON", self.decode_cmd_no_arg),
+            0xF5: ("CMD_HS_TIME_LOTTERY_GUESS", self.decode_cmd_no_arg),
+            0xF6: ("CMD_HS_TOWN_SHOOTING_GALLERY", self.decode_cmd_no_arg),
+            0xF7: ("CMD_HS_UNK_1", self.decode_cmd_no_arg),
+            0xF8: ("CMD_HS_UNK_3_LOWER", self.decode_cmd_no_arg),
+            0xF9: ("CMD_HS_HORSE_BACK_BALLOON", self.decode_cmd_no_arg),
+            0xFA: ("CMD_HS_DEKU_PLAYGROUND_DAY_1", self.decode_cmd_no_arg),
+            0xFB: ("CMD_HS_DEKU_PLAYGROUND_DAY_2", self.decode_cmd_no_arg),
+            0xFC: ("CMD_HS_DEKU_PLAYGROUND_DAY_3", self.decode_cmd_no_arg),
+            0xFD: ("CMD_DEKU_PLAYGROUND_NAME_DAY_1", self.decode_cmd_no_arg),
+            0xFE: ("CMD_DEKU_PLAYGROUND_NAME_DAY_2", self.decode_cmd_no_arg),
+            0xFF: ("CMD_DEKU_PLAYGROUND_NAME_DAY_3", self.decode_cmd_no_arg)
+        }
+        if self.decodedText == "":
+            prevText = False
+            prevNewline = False
+            self.decodePos = 0
+            textLen = len(self.text)
+            while self.decodePos < textLen:
+                char = self.text[self.decodePos]
+                self.decodePos += 1
+
+                if char >= 0x20 and char <= 0xBB: # Characters
+                    if not prevText or prevNewline:
+                        self.decodedText += '"'
+
+                    if char == 0x22: # Handle escaping "
+                        self.decodedText += '\\"'
+                    elif char >= 0xB0: # Button characters
+                        self.decodedText += f'{MessageNES.BUTTONMAP[char]}'
+                    else:
+                        self.decodedText += chr(char)
+
+                    prevText = True
+                    prevNewline = False
+                elif char == 0x11: # New line
+                    if not prevText or prevNewline:
+                        self.decodedText += '"'
+
+                    self.decodedText += f'\\n"\n'
+
+                    prevText = False
+                    prevNewline = True
+                else: # Control Codes (see message_data_fmt_nes.h)
+                    if prevText and not prevNewline:
+                        self.decodedText += '" '
```

Comment:
I noted there's a trailing space in some lines of the extracted text.
This line and the `macro` function seem to be the culprits.

Example:
![image](https://github.com/zeldaret/mm/assets/7416381/3e04d423-9524-479b-bb90-448a12cb79f6)

Not sure if it is worth go over the trouble of fixing this since it is a so small issue

Reply:
Not as simple as just the one line (had more to do with the assumption that each cmd always ended with a space). Though I think I prefer the new algorithm anyways so thanks for bringing it up.